src/query/arithmetic.h

```diff
@@ -31,7 +31,13 @@
 #include "object_domain.h"
 #include "thread.h"
 
+#if defined (__cplusplus)
```

Comment:
since dependent functions are not wrapped in defined (__cplusplus), includes shouldn't be either.